### DISCUSSION TITLE
Issue building against slurm version 23.02.7

### DISCUSSION THREAD

**Discussion by ajneil-gh** [PySlurm/pyslurm#336] [General]
Hi I am trying to build pyslurm on our cluster.  We have slurm version 23.02.7 installed.  I checked out the repo with:
git clone --branch=v23.2.2 https://github.com/PySlurm/pyslurm.git

set the SLURM_INCLUDE_DIR and SLURM_LIB_DIR variables and then ran the scripts/build.sh.

However, I receive this error which has me stumped atm:

/opt/sw/spack/apps/linux-rhel8-x86_64_v2/gcc-9.3.0/gcc-10.3.0-ya/bin/gcc -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -fPIC -fPIC -Ipyslurm -I/opt/slurm/current/include -I. -I/opt/sw/spack/apps/linux-rhel8-x86_64_v2/gcc-10.3.0/python-3.9.9-jh/include/python3.9 -c pyslurm/pyslurm.c -o build/temp.linux-x86_64-3.9/pyslurm/pyslurm.o
pyslurm/pyslurm.c: In function '__pyx_pymod_exec_pyslurm':
pyslurm/pyslurm.c:90536:40: error: 'KILL_JOB_ARRAY' undeclared (first use in this function); did you mean 'KILL_JOB_RESV'?
90536 |   __pyx_t_2 = __Pyx_PyInt_From_uint8_t(KILL_JOB_ARRAY); if (unlikely(!__pyx_t_2)) __PYX_ERR(7, 412, __pyx_L1_error)
      |                                        ^~~~~~~~~~~~~~
      |                                        KILL_JOB_RESV
pyslurm/pyslurm.c:90536:40: note: each undeclared identifier is reported only once for each function it appears in

I'd appreciate any pointers.

-Alastair

**Comment by tazend** [comment]
Hi @ajneil-gh 

you might want to try building from the [23.2.x branch](https://github.com/PySlurm/pyslurm/tree/23.2.x) directly. This branch is a few commits ahead of the 23.2.2 tag, and includes a fix for this problem. (I'll try to tag another release for 23.02 soon though)

**Reply by ajneil-gh** [reply]
Thanks I'll give that a shot
